docs/02_database/05_database-interface/01_entitydb.mdx

```diff
@@ -662,62 +747,417 @@ final var subscription = db.subscribe(Trade.class)
+    updateFrequency = PalDuration.Every(Duration.ofMinutes(5))
+)
+```
+
+</TabItem>
+<TabItem value="java">
+
+```java
+// single-value range
+var range = db.dynamicRangeSubscribeBuilder(
+        () -> Trade.byDate(DateTime.now().withTimeAtStartOfDay())
+    ).withUpdateFrequency(new PalDuration.At(LocalTime.MIDNIGHT))
+        .toFlowable();
+// interval range
+var range = db.dynamicRangeSubscribeBuilder(
+        () -> Trade.byDate(DateTime.now().minusHours(1)),
+        () -> Trade.byDate(DateTime.now().plusHours(1))
+    ).withUpdateFrequency(new PalDuration.Every(Duration.ofMinutes(5)))
+        .toFlowable();
+```
+
+</TabItem>
+</Tabs>
+
+#### Parameters
+
+For static ranges `rangeSubscribe` supports the parameters below.
+
+In the Java API, the function returns a builder, where optional parameters can be set.
+
+| Name             | Type             | Required            | Meaning                                                                      | Default value        |
+|------------------|------------------|---------------------|------------------------------------------------------------------------------|----------------------|
+| Start index      | `EntityIndex<E>` | ✔️                  | The index entry to read from                                                 |                      |
+| End index        | `EntityIndex<E>` | For interval ranges | The index entry to read to                                                   |                      |
+| numKeyFields     | `Int`            | ❌️                  | The number of key fields to take into account for the range                  |                      |
+| updateFrequency  | `PalDuration`    | For dynamic ranges  | A schedule to update dynamic range boundaries                                | ?                    |
+| delay            | `Int`            | ❌                  | Group and publish updates every x ms                                         | 200ms                |
+| fields           | `Set<String>`    | ❌                  | Only listen to changes on selected fields (filters `ModifyResult` on fields) | listen to all fields |
+| subscribeLocally | `Boolean`        | ❌                  | When in a cluster, only listen to local updates                              | false                |
+| backwardJoins    | `Boolean`        | ❌                  | subscribe to changes on sub-tables (backwards joins)                         | false                |
+
+#### Example
+<Tabs defaultValue="kotlin" values={[{ label: 'Kotlin', value: 'kotlin', }, { label: 'Java', value: 'java', }]}>
+<TabItem value="kotlin">
+
+```kotlin
+val range = asyncEntityDb.rangeSubscribe(
+    from = Trade.byDate(startDate),
+    to = Trade.byDate(endDate),
+    numKeyFields = 1,
+    delay = 500,
+    fields = emptySet(),
+    subscribeLocally = true,
+    backwardJoins = true
+)
+
+```
+
+</TabItem>
+<TabItem value="java">
+
+```java
+val range = db.staticRangeSubscribeBuilder(
```

Comment:
do we need samples for dynamic range subscriber builder?

Reply:
we do have some samples for the `dynamicRangeSubscribeBuilder`?

```java
// single-value range
var range = db.dynamicRangeSubscribeBuilder(
        () -> Trade.byDate(DateTime.now().withTimeAtStartOfDay())
    ).withUpdateFrequency(new PalDuration.At(LocalTime.MIDNIGHT))
        .toFlowable();
// interval range
var range = db.dynamicRangeSubscribeBuilder(
        () -> Trade.byDate(DateTime.now().minusHours(1)),
        () -> Trade.byDate(DateTime.now().plusHours(1))
    ).withUpdateFrequency(new PalDuration.Every(Duration.ofMinutes(5)))
        .toFlowable();
```

Reply:
Yes, I just meant that in the parameters sections we only have samples for non dynamic use cases. Maybe we don't need them here, that is fine.